packages/vite/src/node/optimizer/scan.ts

```diff
@@ -85,34 +84,28 @@ export async function scanImports(
     debug(`Crawling dependencies using entries:\n  ${entries.join('\n  ')}`)
   }
 
-  const tempDir = path.join(config.cacheDir!, 'temp')
   const deps: Record<string, string> = {}
   const missing: Record<string, string> = {}
   const container = await createPluginContainer(config)
   const plugin = esbuildScanPlugin(config, container, deps, missing, entries)
 
+  const { plugins = [], ...esbuildOptions } =
+    config.optimizeDeps?.esbuildOptions ?? {}
+
   await Promise.all(
     entries.map((entry) =>
       build({
+        write: false,
         entryPoints: [entry],
         bundle: true,
         format: 'esm',
         logLevel: 'error',
-        outdir: tempDir,
-        plugins: [plugin]
+        plugins: [...plugins, plugin],
+        ...esbuildOptions
       })
     )
   )
 
-  try {
-    emptyDir(tempDir)
-    fs.rmdirSync(tempDir)
-  } catch (err) {
-    if (err.code !== 'ENOENT') {
-      throw err
-    }
-  }
-
```

Comment:
Nice! This is great, but I think we should push it to another PR. Just in case there are issues, I think this PR should only be about adding the esbuild options, so we can better track this with git blame in case there are regressions. The refactoring to log deprecated options also looks good, I'm debating if we should also do it in another PR (maybe this one is fine)

Reply:
Yeah, fair enough, removed the tmpdir stuff. I think the log stuff is fine :)

Reply:
=> #3231